sys/riscv/riscv/pmap.c

```diff
@@ -2666,8 +2707,7 @@ pmap_enter(pmap_t pmap, vm_offset_t va, vm_page_t m, vm_prot_t prot,
 #if __has_feature(capabilities)
 	if (prot & VM_PROT_READ_CAP)
 		new_l3 |= PTE_CR;
-	if (prot & VM_PROT_WRITE_CAP)
-		new_l3 |= PTE_CW | PTE_CD;
+	new_l3 |= cheri_pte_cw(m, prot);
```

Comment:
Should this not mirror W and D, ie set CW here if VM_PROT_WRITE_CAP still, and move the CD setting down to below where the MANAGED handling is. It should also just be inlined, the code isn't reused.

Reply:
Also (a) why do we need to set CD but D is fine left unset (b) why are we not handling VPO_UNMANAGED specially (c) why do we need a separate PGA_CAPSTORE when we already have prot?

Reply:
I'll address the other points in turn, but (c) deserves its own answer.  For some definition of capdirty tracking, you're right, we'd not need a separate flag.  However, protection is associated with page *aliases*, so something associated with the page itself is nice to have, and also we want, eventually, to have the revoker not need to look at all pages whose maxprot *permits* capability store but rather just those that aren't known not to have capabilities on them.

Thus, we refine cap-dirty into three states:

* has capabilities, at least as of last we looked (both `PGA_CAP{DIRTY,STORE}` asserted)

* did not have capabilities when last we looked, but we may have looked asynchronously, so there may be some we didn't notice (`PGA_CAPDIRTY` clear and `PGA_CAPDIRTY` asserted)

* does not have capabilities (never had or we have checked in a properly defensive way; both `PGA_CAP{DIRTY,STORE}` clear)

This is spelled out in excessive detail in the new revocation tech report, but it's somewhat hidden in this PR.  `PGA_CAPSTORE` leads a pretty boring life, here, but it should get more interesting.